gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/Topic.java

```diff
@@ -0,0 +1,162 @@
+import java.io.ObjectInputStream;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.Future;
+
+/**
+ * PubSub Topic.
+ */
+public class Topic extends TopicInfo {
+
+  private static final long serialVersionUID = -2686692223763315944L;
+
+  private final PubSubOptions options;
+  private transient PubSub pubsub;
+
+  public static final class Builder extends TopicInfo.Builder {
+
+    private final PubSub pubsub;
+    private final BuilderImpl delegate;
+
+    private Builder(Topic topic) {
+      pubsub = topic.pubsub;
+      delegate = new BuilderImpl(topic);
+    }
+
+    @Override
+    public Builder name(String name) {
+      delegate.name(name);
+      return this;
+    }
+
+    @Override
+    public Topic build() {
+      return new Topic(this.pubsub, this.delegate);
+    }
+  }
+
+  Topic(PubSub pubsub, BuilderImpl builder) {
+    super(builder);
+    this.pubsub = checkNotNull(pubsub);
+    options = pubsub.options();
+  }
+
+  @Override
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(options, super.hashCode());
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null || getClass() != obj.getClass()) {
+      return false;
+    }
+    Topic other = (Topic) obj;
+    return Objects.equals(toPb(), other.toPb()) && Objects.equals(options, other.options);
+  }
+
+  public PubSub pubSub() {
+    return pubsub;
+  }
+
+  public boolean delete() {
+    return pubsub.deleteTopic(name());
+  }
+
+  public Future<Boolean> deleteAsync() {
+    return pubsub.deleteTopicAsync(name());
+  }
+
+  public Topic reload() {
+    return pubsub.getTopic(name());
+  }
+
+  public Future<Topic> reloadAsync() {
+    return pubsub.getTopicAsync(name());
+  }
+
+  public String publish(Message message) {
+    return pubsub.publish(name(), message);
+  }
+
+  public Future<String> publishAsync(Message message) {
+    return pubsub.publishAsync(name(), message);
+  }
+
+  public List<String> publish(Message message, Message... messages) {
+    return pubsub.publish(name(), message, messages);
+  }
+
+  public Future<List<String>> publishAsync(Message message, Message... messages) {
+    return pubsub.publishAsync(name(), message, messages);
+  }
+
+  public List<String> publish(Iterable<Message> messages) {
+    return pubsub.publish(name(), messages);
+  }
+
+  public Future<List<String>> publishAsync(Iterable<Message> messages) {
+    return pubsub.publishAsync(name(), messages);
+  }
+
+  public Page<Subscription> listSubscriptions(ListOption... options) {
+    return pubsub.listSubscriptions(name(), options);
+  }
+
+  public Future<AsyncPage<Subscription>> listSubscriptionsAsync(ListOption... options) {
+    return pubsub.listSubscriptionsAsync(name(), options);
+  }
+
+  private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid prefixing parameters by in, out or inOut. Uses Javadoc to document this behavior.](https://www.codacy.com/app/mziccard/gcloud-java/file/2397650225/issues/source?bid=3123109&fileBranchId=3188953#l144)